packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -765,6 +816,23 @@ const DatePickerInput = (props) => {
     { 'is-invalid': isInvalid },
   ]);
 
+  // Indicates selected date from calendar popup for SR
+  let inputDate;
+  if (DateUtil.isValidDate(value, momentDateFormat)) {
+    inputDate = `${getLocalizedDateForScreenReader(DateUtil.createSafeDate(dateValue, initialTimeZone), { intl, locale: intl.locale })}`;
+  }
+  let calendarDate = inputDate ? `${inputDate} ${intl.formatMessage({ id: 'Terra.datePicker.selected' })}` : '';
+
+  // Check if date is excluded or out of range or not included or filtered
+  let invalidEntry = '';
+  if (DateUtil.isDateExcluded(DateUtil.createSafeDate(dateValue, initialTimeZone), props.excludeDates)
+      || DateUtil.isDateOutOfRange(DateUtil.createSafeDate(dateValue, initialTimeZone), DateUtil.createSafeDate(DateUtil.getMinDate(props.minDate), initialTimeZone), DateUtil.createSafeDate(DateUtil.getMaxDate(props.maxDate), initialTimeZone))
+      || DateUtil.isDateNotIncluded(DateUtil.createSafeDate(dateValue, initialTimeZone), props.includeDates)
+      || (props.filterDate && !props.filterDate(DateUtil.createSafeDate(dateValue, initialTimeZone)))) {
+    invalidEntry = intl.formatMessage({ id: 'Terra.datePicker.invalidDate' });
```

Comment:
what happens if we call `handleInvalidInputChange` instead of this here

Reply:
Used `invalidEntry`  to allow invalid date to be notified as and when the complete date is entered.  `handleInvalidInputChange`  does not take effect in this case, in spite of being re-written.